membership-attribute-service/app/services/AttributesMaker.scala

```diff
@@ -70,12 +98,37 @@ class AttributesMaker extends LazyLogging {
           MembershipJoinDate = zuora.MembershipJoinDate,
           DigitalSubscriptionExpiryDate = zuora.DigitalSubscriptionExpiryDate,
           MembershipNumber = None,
-          AdFree = None
+          AdFree = None,
+          ActionAvailableFor = zuora.ActionAvailableFor
         ))
       case (None, _) => None
     }
   }
 
+  def actionAvailableFor(accountSummary: AccountSummary, subscription: Subscription[AnyPlan],
+                         paymentMethodGetter: PaymentMethodId => Future[String \/ PaymentMethodResponse])(implicit ec: ExecutionContext): Future[Boolean] = {
+
+    def creditCard(paymentMethodResponse: PaymentMethodResponse) = paymentMethodResponse.paymentMethodType == "CreditCardReferenceTransaction" || paymentMethodResponse.paymentMethodType == "CreditCard"
+
+    val stillFreshInDays = 27
+    def recentEnough(lastTransationDateTime: DateTime) = lastTransationDateTime.plusDays(stillFreshInDays).isAfterNow
+
+    def actionForSub(accountSummary: AccountSummary, sub: Subscription[AnyPlan]): Future[String \/ Boolean] = {
+      if(accountSummary.balance > 0 && accountSummary.defaultPaymentMethod.isDefined) {
+        val eventualPaymentMethod: Future[\/[String, PaymentMethodResponse]] = paymentMethodGetter(accountSummary.defaultPaymentMethod.get.id)
+        eventualPaymentMethod map { maybePaymentMethod: \/[String, PaymentMethodResponse] =>
+          maybePaymentMethod.map { pm: PaymentMethodResponse =>
+            if(creditCard(pm) && pm.numConsecutiveFailures > 0 && recentEnough(pm.lastTransactionDateTime))
```

Comment:
Could extract this into a val called _userInPaymentFailure_?

Reply:
Yep, that's a better idea. Will do.